lib/rules/no-return-value-from-each-iteratee.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @fileoverview Rule to check the iteratee in a call to _.each does not return a value
+ */
+'use strict';
+
+var _ = require('lodash');
+var lodashUtil = require('../util/lodashUtil');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    return {
+        ReturnStatement: 
+        function (node) {
+            function getSurroundingFunction(astNode) {
+                var ancestors = context.getAncestors();
+                var current = astNode;
+                while (!_.isUndefined(current)) {
+                    if (current.type === 'FunctionExpression' ||
+                        current.type === 'FunctionDeclaration') {
+                        break;
+                    }
+                    current = ancestors.pop();
+                }
+                return current;
+            }
+
+            function isReturnFromIteratee(astNode) {
+                var surroundingFunction = getSurroundingFunction(astNode);
+                return !_.isUndefined(surroundingFunction) &&
+                    lodashUtil.isCallToMethod(surroundingFunction.parent, 'each') &&
+                    lodashUtil.getLodashIteratee(surroundingFunction.parent) === surroundingFunction;
```

Comment:
Was mostly paranoia, but it turns out that if you remove the last check, this test fails:

```_.each(function(x) { return x+1;}, function(f) { console.log (f);});```

Granted, that's some dumb code and should probably get some kind of warning, but not from this rule. :)